taichi/backends/metal/metal_program.h

```diff
@@ -36,6 +36,9 @@ class MetalProgramImpl : public ProgramImpl {
     metal_kernel_mgr_->synchronize();
   }
 
+  virtual void destroy_snode_tree(SNodeTree *snode_tree) override{
```

Comment:
For derived class it's okay to use `override` only ;). `virtual` is only necessary for base class. ;)